libpyvinyl/Parameters/Parameter.py

```diff
@@ -8,126 +8,168 @@ class Parameter(AbstractBaseClass):
+                    "         This option is declared differently w.r.t. to the previous ones"
+                )
+                # should it throw an expection?
+                raise ValueError("Parameter", "options", "multiple validities")
+
         if isinstance(option, list):
             self.options += option
         else:
             self.options.append(option)
 
-    def set_value(self, value):
+    def set_option_as_legal(self, mybool=True):
+        self.options_are_legal = mybool
+
+    @property
+    def value(self):
+        return self._value
+
+    @value.setter
+    def value(self, value):
         """
         Sets value of this parameter if value is legal, otherwise warning is shown
 
         This could be expanded to raise an exception, or such could be in is_legal
         """
         if self.is_legal(value):
-            self.value = value
+            self._value = value
         else:
-            print("WARNING: Value of parameter '" + self.name
-                  + "' illegal, ignored.")
+            raise ValueError("Value of parameter '" + self.name + "' illegal.")
 
-    def is_legal(self, value=None):
+    def is_legal(self, values=None):
         """
         Checks whether or not given or contained value is legal given constraints.
 
         Illegal intervals have the highest priority to be checked. Then it will check the
         legal intervals and options. The overlaps among the constrains will be overridden by
         the constrain of higher priority.
         """
-        if value is None:
-            value = self.value
-
-        # Check illegal intervals
-        for illegal_interval in self.illegal_intervals:
-            if illegal_interval[0] < value < illegal_interval[1]:
-                return False
-
-        # Check legal intervals
-        is_inside_a_legal_interval = False
-        for legal_interval in self.legal_intervals:
-            if legal_interval[0] < value < legal_interval[1]:
-                is_inside_a_legal_interval = True
 
-        if not is_inside_a_legal_interval and len(self.legal_intervals) > 0:
-            return False
+        truths = []
+        if values is None:
+            values = self._value
 
-        # checked intervals, can return if options not used (frequent case)
-        if len(self.options) == 0:
-            return True
+        if not hasattr(values, "__iter__") or isinstance(values, str):
+            #            print(str(hasattr(values, "__iter__")) + str(values))
+            value = values
 
-        for option in self.options:
-            if option == value:
-                # If the value matches any option, it is legal
-                return True
+            for option in self.options:
+                if option == value:
+                    #
+                    return self._options_are_legal
+
+            # Check illegal intervals
+            for interval in self.intervals:
+                if interval[0] < value < interval[1]:
```

Comment:
I think we should probably use <= to include the end points in the interval.

Reply:
As a user I would find that more confusing as end points are not handled in the same way for each end of the interval. I saw elsewhere that using options to include the end points is a possible workaround, but think the most common use case is a closed interval.

Reply:
You are right. The [a,b) kind of interval are useful when you are segmenting an interval in bins, but of course this is not the case.
I will make closed intervals the default. 
I will also open an issue for supporting open intervals and mark it as enhancement, for a possible future development.

Reply:
ok